### DISCUSSION TITLE
already replaced this function

### DISCUSSION THREAD

**Discussion by 350030173** [vfsfitvnm/frida-il2cpp-bridge#93] [Q&A]
always have this error
**_the frida-il2cpp-bridge version is 0.7.3_**

Error: already replaced this function
    at value (frida/runtime/core.js:315)
    at set implementation (node_modules/frida-il2cpp-bridge/dist/il2cpp/structs/method.js:122)

    at build (node_modules/frida-il2cpp-bridge/dist/il2cpp/tracer.js:207)

**_i use frida-il2cpp-bridge version is 0.6.6 dont have this error_**

**Comment by vfsfitvnm** [comment]
What's your code snippet?

**Reply by vfsfitvnm** [reply]
Oh yeah I'm retarded, I deleted the line of code where I restored the original implementation

**Reply by vfsfitvnm** [reply]
However this is also why I want to do this: https://github.com/vfsfitvnm/frida-il2cpp-bridge#planned

**Reply by 350030173** [reply]
> What's your code snippet?

when attach , the code is :
Il2Cpp.trace()
                    .assemblies(Il2Cpp.Domain.assembly("Assembly-CSharp"))
                    .commit()
                    .detailed() 
                    .build();

**Reply by 350030173** [reply]
> However this is also why I want to do this: https://github.com/vfsfitvnm/frida-il2cpp-bridge#planned

i hope dont remove **Il2Cpp.Tracer::detailed mode.** , This feature is very useful，i always use this feature  to see the args value and return value.(my english is pool)

**Reply by vfsfitvnm** [reply]
@350030173 
Do you still need to see the call graph when using `Il2Cpp.Tracer::detailed`? E.g:
**Current**
```
il2cpp: 0x00d0d6b0 ┌─System.String.EndsWith[this = "hello"](System.String value = "test", System.StringComparison comparisonType = CurrentCulture);
il2cpp: 0x00d0ae60 │ ┌─System.String.get_Chars[this = "hello"](System.Int32 index = 0);
il2cpp: 0x00d0ae60 │ └─System.String.get_Chars = 104;
il2cpp: 0x00d0ae60 │ ┌─System.String.get_Chars[this = "hello"](System.Int32 index = 0);
il2cpp: 0x00d0ae60 │ └─System.String.get_Chars = 104;
il2cpp: 0x00d0ae60 │ ┌─System.String.get_Chars[this = "hello"](System.Int32 index = 0);
il2cpp: 0x00d0ae60 │ └─System.String.get_Chars = 104;
il2cpp: 0x00d0d6b0 └─System.String.EndsWith = false;
```
**Planned**
```
il2cpp: 0x00d0ae60 System.String.get_Chars[this = "hello"](System.Int32 index = 0) = 104;
il2cpp: 0x00d0ae60 System.String.get_Chars[this = "hello"](System.Int32 index = 0) = 104;
il2cpp: 0x00d0ae60 System.String.get_Chars[this = "hello"](System.Int32 index = 0) = 104;
il2cpp: 0x00d0d6b0 System.String.EndsWith[this = "hello"](System.String value = "test", System.StringComparison comparisonType = CurrentCulture) = false;
```
I find the current one difficult to read, don't you? What's your opinion?

**Reply by 350030173** [reply]
> @350030173 Do you still need to see the call graph when using `Il2Cpp.Tracer::detailed`? E.g: **Current**
> 
> ```
> il2cpp: 0x00d0d6b0 ┌─System.String.EndsWith[this = "hello"](System.String value = "test", System.StringComparison comparisonType = CurrentCulture);
> il2cpp: 0x00d0ae60 │ ┌─System.String.get_Chars[this = "hello"](System.Int32 index = 0);
> il2cpp: 0x00d0ae60 │ └─System.String.get_Chars = 104;
> il2cpp: 0x00d0ae60 │ ┌─System.String.get_Chars[this = "hello"](System.Int32 index = 0);
> il2cpp: 0x00d0ae60 │ └─System.String.get_Chars = 104;
> il2cpp: 0x00d0ae60 │ ┌─System.String.get_Chars[this = "hello"](System.Int32 index = 0);
> il2cpp: 0x00d0ae60 │ └─System.String.get_Chars = 104;
> il2cpp: 0x00d0d6b0 └─System.String.EndsWith = false;
> ```
> 
> **Planned**
> 
> ```
> il2cpp: 0x00d0ae60 System.String.get_Chars[this = "hello"](System.Int32 index = 0) = 104;
> il2cpp: 0x00d0ae60 System.String.get_Chars[this = "hello"](System.Int32 index = 0) = 104;
> il2cpp: 0x00d0ae60 System.String.get_Chars[this = "hello"](System.Int32 index = 0) = 104;
> il2cpp: 0x00d0d6b0 System.String.EndsWith[this = "hello"](System.String value = "test", System.StringComparison comparisonType = CurrentCulture) = false;
> ```
> 
> I find the current one difficult to read, don't you? What's your opinion?

**Current:**
have call graph,can get a better sense of priorities,
can know which function calls which function,

**Planned:**
More concise

**Suggestion:**
Or whether you can keep "Current" and add "Planned" ,

**Reply by vfsfitvnm** [reply]
Ok, I'll.keep the planned one. At this point I'll just do minor refactor, e.g. make simple, full tracing use Interceptor.attach for better performance
